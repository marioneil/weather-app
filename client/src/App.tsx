import { useState } from "react";
// import reactLogo from "./assets/react.svg";
// import viteLogo from "/vite.svg";
// import "./App.css";
// import "bootstrap/dist/css/bootstrap.css";
import "./customBootstrap.scss";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Layout } from "./component/layout/Layout";
import { Today } from "./component/cards/Today";
import { WeatherProvider } from "./context/WeatherContext";

function App() {
  return (
    <WeatherProvider>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route path="/today" element={<Today />} />
            <Route path="/about" element={<h1>Abbout</h1>} />
          </Route>
        </Routes>
      </BrowserRouter>
    </WeatherProvider>
  );
}

export default App;
