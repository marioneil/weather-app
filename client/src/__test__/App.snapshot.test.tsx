// App.snapshot.test.js
import { create } from "react-test-renderer";

import App from "../App";

test("renders learn react link", () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
