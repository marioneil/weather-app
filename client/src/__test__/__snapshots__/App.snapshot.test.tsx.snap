// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <nav
    className="navbar bg-primary"
    data-bs-theme="dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        data-testid="Navbar.Header"
        href="#"
      >
        The Weather Channel1
      </a>
      <form
        className="d-flex"
        role="search"
      >
        <div
          className="position-relative "
        >
          <input
            aria-label="Search"
            className="form-control me-2 bg-primary border-light"
            onChange={[Function]}
            placeholder="Search"
            style={
              {
                "width": "20rem",
              }
            }
            type="search"
            value=""
          />
          <div
            className="list-group position-absolute"
          />
        </div>
      </form>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              className="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              role="button"
            >
              Dropdown
            </a>
            <ul
              className="dropdown-menu"
            >
              <li>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <hr
                  className="dropdown-divider"
                />
              </li>
              <li>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </li>
            </ul>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-disabled="true"
              className="nav-link disabled"
            >
              Disabled
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="d-flex bg-primary-subtle"
  >
    <div
      className="d-flex align-items-center justify-content-center py-2 flex-grow-1 border-end  border-white"
    >
      <div
        className="flex-shrink-0"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 19h-11c-2.206 0-4-1.794-4-4 0-1.861 1.277-3.429 3.001-3.874l-.001-.126c0-3.309 2.691-6 6-6 2.587 0 4.824 1.638 5.65 4.015 2.942-.246 5.35 2.113 5.35 4.985 0 2.757-2.243 5-5 5zm-11.095-6.006c-1.008.006-1.905.903-1.905 2.006s.897 2 2 2h11c1.654 0 3-1.346 3-3s-1.346-3-3-3c-.243 0-.5.041-.81.13l-1.075.307-.186-1.103c-.325-1.932-1.977-3.334-3.929-3.334-2.206 0-4 1.794-4 4 0 .272.027.545.082.811l.244 1.199-1.421-.016z"
          />
        </svg>
      </div>
      <div
        className="ms-3"
        data-testid="navbar2"
      >
        Â° C
        , 
        London,  City of London, Greater London, United Kingdom 
      </div>
    </div>
    <div
      className="flex-grow-1 ms-3"
    >
       
    </div>
  </div>
  <ul
    className="nav justify-content-between bg-secondary text-bg-dark "
    data-bs-theme="dark"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/today"
        onClick={[Function]}
      >
        Today
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/hourly"
        onClick={[Function]}
      >
        Hourly
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
      >
        10 Day
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        aria-disabled="true"
        className="nav-link disabled"
      >
        Monthly
      </a>
    </li>
  </ul>
</div>
`;
